ext/scheduler/airflow2/airflow.go

```diff
@@ -417,8 +334,8 @@ func JobNameFromPath(filePath, suffix string) string {
 
 func NewScheduler(bucketFac BucketFactory, httpClient HTTPClient, compiler models.JobCompiler) *scheduler {
 	return &scheduler{
-		bucketFac:  bucketFac,
-		compiler:   compiler,
-		httpClient: httpClient,
+		bucketFac: bucketFac,
+		compiler:  compiler,
+		client:    newClient(httpClient),
```

Comment:
this newClient method is not adding any value add imo, we can just initialize as a struct.

Reply:
initialising directly